elide-core/src/main/java/com/yahoo/elide/core/sort/Sorting.java

```diff
@@ -68,20 +64,34 @@ public Sorting(final Map<String, SortOrder> sortingRules) {
      * @return The valid sorting rules - validated through the entity dictionary, or empty dictionary
      * @throws InvalidValueException when sorting values are not valid for the jpa entity
      */
-    public <T> Map<String, SortOrder> getValidSortingRules(final Class<T> entityClass,
-                                                           final EntityDictionary dictionary)
+    public <T> Map<Path, SortOrder> getValidSortingRules(final Class<T> entityClass,
+                                                         final EntityDictionary dictionary)
             throws InvalidValueException {
-        hasValidSortingRules(entityClass, dictionary);
-        return replaceIdRule(dictionary.getIdFieldName(entityClass));
-    }
+        Map<Path, SortOrder> returnMap = new LinkedHashMap<>();
+        for (Map.Entry<String, SortOrder> entry : replaceIdRule(dictionary.getIdFieldName(entityClass)).entrySet()) {
```

Comment:
If we're concerned about performance of code restructuring, we could add one more private method here;

`Stream<Path> getSortingRules(Class<T>, EntityDictionary)`

This way we can stay DRY on our `replaceIdRule` code (and lazily evaluate) and still only have to iterate once since streams are lazy.